src/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -0,0 +1,188 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.analysis;
+
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+import picard.PicardException;
+import picard.util.MathUtil;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.*;
+
+/**
+ * Created by David Benjamin on 5/13/15.
+ */
+public class TheoreticalSensitivity {
+    /*
+    @param the probability of depth n is depthDistribution[n] for n = 0, 1. . . N - 1
+    @param the probability of quality q is qualityDistribution[q] for q = 0, 1. . . Q
+    @param sample size is the number of random sums of quality scores for each m
+    @param logOddsThreshold is the log_10 of the likelihood ratio required to call a SNP,
+    for example 5 if the variant likelihood must be 10^5 times greater
+     */
+    private static final Log log = Log.getInstance(TheoreticalSensitivity.class);
+    private static final int SAMPLING_MAX = 600; //prevent 'infinite' loops
+
+    public static double hetSNPSensitivity(final double[] depthDistribution, final double[] qualityDistribution,
+                                  final int sampleSize, final double logOddsThreshold) {
+        final int N = depthDistribution.length;
+
+        log.info("Creating Roulette Wheel");
+        final RouletteWheel qualitySampler = new RouletteWheel(qualityDistribution);
+
+        //qualitySums[m] is a random sample of sums of m quality scores, for m = 0, 1, N - 1
+        log.info("Calculating quality sums from quality sampler");
+        final List<ArrayList<Integer>> qualitySums = qualitySampler.sampleCumulativeSums(N, sampleSize);
+
+        //if a quality sum of m qualities exceeds the quality sum threshold for n total reads, a SNP is called
+        final ArrayList<Double> qualitySumThresholds = new ArrayList<Double>(N);
+        for (int n = 0; n < N; n++) qualitySumThresholds.add(10*(n*Math.log10(2) + logOddsThreshold));
```

Comment:
may not affect performance in reality, but Math.log10(2) is a constant, id make that a final double so it isnt recalculated unnecessarily